immuni_exposure_ingestion/models/validators.py

```diff
@@ -7,58 +9,152 @@
             raise ValidationError(
-                f"Too many TEKs. (actual: {n_keys}, max_allowed: {config.MAX_KEYS_PER_UPLOAD})."
+                f"Too many TEKs. (actual: {n_teks}, max_allowed: {config.MAX_KEYS_PER_UPLOAD})."
             )
 
-        sorted_teks = sorted(value, key=operator.attrgetter("rolling_start_number"))
+        valid_teks = self._filter_valid_teks(teks)
+
+        sorted_teks = sorted(
+            valid_teks, key=operator.attrgetter("rolling_start_number")
+        )
         rolling_start_numbers = [t.rolling_start_number for t in sorted_teks]
 
+        # TEKs with the same rolling_start_number can exist when considering a rolling_period < 144.
+        # Though, since the latter are ignored, we check the rolling_start_number is unique.
         if len(rolling_start_numbers) != len(set(rolling_start_numbers)):
             raise ValidationError("Rolling start numbers are not unique.")
 
         next_rolling_start_number = sorted_teks[0].rolling_start_number
         for current in sorted_teks:
             if current.rolling_start_number < next_rolling_start_number:
                 raise ValidationError("Overlapping rolling start numbers.")
-            next_rolling_start_number = current.rolling_start_number + current.rolling_period
+            next_rolling_start_number = (
+                current.rolling_start_number + current.rolling_period
+            )
 
         if config.ALLOW_NON_CONSECUTIVE_TEKS:
             return
 
+        # TODO: Change this logic once ignored_teks are not ignored anymore.
         initial_start_number = rolling_start_numbers[0]
         if rolling_start_numbers != [
             initial_start_number + 144 * i for i in range(len(rolling_start_numbers))
         ]:
             raise ValidationError("Some TemporaryExposureKeys are missing.")
+
+    @staticmethod
+    def _today_rolling_start_number() -> int:
+        return int(
+            datetime.utcnow()
+            .replace(hour=0, minute=0, second=0, microsecond=0, tzinfo=timezone.utc)
+            .timestamp()
+            / timedelta(minutes=10).total_seconds()
+        )
+
+    def _filter_valid_teks(
+        self, teks: List[TemporaryExposureKey]
+    ) -> List[TemporaryExposureKey]:
+        """
+        From all given TEKs, return the list of valid ones.
+        Some may be valid for the final framework, but currently ignored.
+        Raise as soon as an invalid TEKs is found.
+
+        :param teks: the list of TEKs to be filtered.
+        :raises: ValidationError in case at least one TEK is deemed invalid.
+        :return: the list of valid TEKs.
+        """
+
+        valid_teks = ignored_teks = today_teks = list()
+        today_rolling_start_number = self._today_rolling_start_number()
+
+        for tek in teks:
+
+            if tek.rolling_period == _ROLLING_PERIOD_MAX:
+                valid_teks.append(tek)
+            elif not _ROLLING_PERIOD_MIN <= tek.rolling_period < _ROLLING_PERIOD_MAX:
```

Comment:
is a rolling period of `0` a valid one? isn't it weird?

Reply:
It's indeed strange since it would mean the TEK is valid for 0 time.

Reply:
I couldn't find the definition [1,144] I remember, but only "positive number below 144", and I ofc ended up on the wrong side of 0. 😂